master/pkg/searcher/grid.go

```diff
@@ -36,85 +36,77 @@ func newGridSearch(config expconf.GridConfig) SearchMethod {
-	trialProgress map[model.RequestID]PartialUnits,
-	trialsClosed map[model.RequestID]bool,
+	runProgress map[int32]float64,
+	runsClosed map[int32]bool,
 ) float64 {
-	if s.MaxConcurrentTrials() > 0 && s.PendingTrials > s.MaxConcurrentTrials() {
+	if s.MaxConcurrentTrials() > 0 && s.PendingRuns > s.MaxConcurrentTrials() {
 		panic("pending trials is greater than max_concurrent_trials")
 	}
-	// XXX
 	// Progress is calculated as follows:
 	//   - InvalidHP trials contribute max_length units since they represent one config within the grid
 	//     and are not replaced with a new config as with random search
 	//   - Other early-exit trials contribute max_length units
 	//   - In progress trials contribute units trained
-	unitsCompleted := 0.
-	// trialsClosed includes InvalidHP trials and other exited trials
-	for range trialsClosed {
-		unitsCompleted += float64(s.MaxLength().Units)
-	}
-	// trialProgress records units trained for all trials except for InvalidHP trials.
-	// This can overlap with trialsClosed so we need to be sure to not double count.
-	for k, v := range trialProgress {
-		if !trialsClosed[k] {
-			unitsCompleted += float64(v)
+	runProgresses := 0.
+
+	for k, v := range runProgress {
+		if runsClosed[k] {
+			runProgresses += 1.0
+		} else {
+			runProgresses += v
 		}
 	}
-	unitsExpected := s.MaxLength().Units * uint64(s.trials)
-	return unitsCompleted / float64(unitsExpected)
+
+	return runProgresses / float64(len(runProgress))
 }
 
 // trialExitedEarly does nothing since grid does not take actions based on
 // search status or progress.
-func (s *gridSearch) trialExitedEarly(
-	ctx context, requestID model.RequestID, exitedReason model.ExitedReason,
-) ([]Operation, error) {
+func (s *gridSearch) runExitedEarly(
+	ctx context, runID int32, exitedReason model.ExitedReason,
+) ([]Action, error) {
 	return nil, nil
 }
 
-func (s *gridSearch) trialClosed(ctx context, _ model.RequestID) ([]Operation, error) {
-	s.PendingTrials--
-	var ops []Operation
-	if len(s.RemainingTrials) > 0 {
-		params := s.RemainingTrials[len(s.RemainingTrials)-1]
-		s.RemainingTrials = s.RemainingTrials[:len(s.RemainingTrials)-1]
-		create := NewCreate(ctx.rand, params, model.TrialWorkloadSequencerType)
-		ops = append(ops, create)
-		ops = append(ops, NewValidateAfter(create.RequestID, s.MaxLength().Units))
-		ops = append(ops, NewClose(create.RequestID))
-		s.PendingTrials++
+func (s *gridSearch) runClosed(ctx context, _ int32) ([]Action, error) {
+	s.PendingRuns--
+	var actions []Action
+	if len(s.RemainingRuns) > 0 {
+		params := s.RemainingRuns[len(s.RemainingRuns)-1]
+		s.RemainingRuns = s.RemainingRuns[:len(s.RemainingRuns)-1]
+		create := NewCreate(ctx.rand, params)
+		actions = append(actions, create)
+		s.PendingRuns++
 	}
-	return ops, nil
+	return actions, nil
 }
 
-func newHyperparameterGrid(params expconf.Hyperparameters) []HParamSample {
+func NewHyperparameterGrid(params expconf.Hyperparameters) []HParamSample {
```

Comment:
why suddenly export this?

Reply:
was probably doing some funny things at one point. fixed.